addons/edi/models/edi_synchronizer.py

```diff
@@ -370,21 +381,25 @@ def execute(self):
                     vals_list = [rec.target_values(rec._record_values()) for rec in batch]
                     if doc.fail_fast:
                         for rec, vals in zip(batch, vals_list):
-                            rec[target].write(vals)
+                            changed_vals = self.filter_changed_vals(vals, comparator, rec[target])
+                            if changed_vals:
+                                rec[target].write(changed_vals)
                     else:
                         for rec, vals in zip(batch, vals_list):
-                            try:
-                                # We use a savepoint here to handle the case where
-                                # vals violate an api.constrains on the target.
-                                # These constraints are checked after updates are
-                                # sent to the database, so the offending change
-                                # must be rolled back for the affected object.
-                                with self.env.cr.savepoint():
-                                    rec[target].write(vals)
-                            except ValidationError as ex:
-                                rec[target].invalidate_cache()
-                                rec.error = ex.name
-                                _logger.exception("Failed to update for %r, %s", rec, rec.name)
+                            changed_vals = self.filter_changed_vals(vals, comparator, rec[target])
```

Comment:
Elide function already checks which fields have changed during preparation. To avoid performance impact on execute, we could store the list of fields inside elide and use that list of fields during `target_values()` to only update the ones that have changed.

Reply:
Makes sense, will do.